EventEnabledInsurance/setup-elastic-search.sh

```diff
@@ -0,0 +1,233 @@
+        node.data: true
+        node.ingest: true
+        node.attr.attr_name: attr_value
+        node.store.allow_mmap: false
+      podTemplate:
+        metadata:
+          labels:
+            name: $ELASTIC_CR_NAME
+            demo: eei
+        spec:
+          containers:
+            - name: elasticsearch
+              resources:
+                requests:
+                  memory: 4Gi
+                  cpu: 1
+                limits:
+                  memory: 4Gi
+                  cpu: 2
+      count: 1
+EOF
+
+echo -e "\n----------------------------------------------------------------------------------------------------------------------------------------------------------\n"
+
+time=0
+ES_CR_STATE=$(oc get elasticsearch -n $ELASTIC_NAMESPACE $ELASTIC_CR_NAME -o json | jq -r '.status.phase')
+while [ "$ES_CR_STATE" != "Ready" ]; do
+  if [ $time -gt 10 ]; then
+    echo "ERROR: Timed-out waiting for elastic search CR to be ready"
+    echo -e "\n----------------------------------------------------------------------------------------------------------------------------------------------------------\n"
+    exit 1
+  fi
+
+  echo "Elasticsearch current status: '$(oc get elasticsearch -n $ELASTIC_NAMESPACE $ELASTIC_CR_NAME -o json | jq -r '.status.phase')'"
+  echo -e "INFO: The elastic search CR is not yet ready, waiting for upto 10 minutes. Waited ${time} minute(s).\n"
+  time=$((time + 1))
+  sleep 60
+  ES_CR_STATE=$(oc get elasticsearch -n $ELASTIC_NAMESPACE $ELASTIC_CR_NAME -o json | jq -r '.status.phase')
+done
+
+echo -e "INFO: The elastic search CR is now ready:\n"
+oc get elasticsearch -n $ELASTIC_NAMESPACE
+echo -e "\n----------------------------------------------------------------------------------------------------------------------------------------------------------\n"
+
+echo -e "INFO: The pods for the elastic search:\n"
+oc get pods -n $ELASTIC_NAMESPACE --selector='elasticsearch.k8s.elastic.co/cluster-name='${ELASTIC_CR_NAME}''
+
+echo -e "\n----------------------------------------------------------------------------------------------------------------------------------------------------------\n"
+
+echo -e "INFO: The elastic search service:\n"
+oc get service -n $ELASTIC_NAMESPACE $ELASTIC_CR_NAME-es-http
+
+echo -e "\n----------------------------------------------------------------------------------------------------------------------------------------------------------\n"
+
+ELASTIC_PASSWORD=$(oc get secret $ELASTIC_CR_NAME-es-elastic-user -n $ELASTIC_NAMESPACE -o go-template='{{.data.elastic | base64decode}}')
+ELASTIC_USER="elastic"
+echo -e "INFO: Got the password for elastic search..."
+
+echo -e "\nINFO: Creating secret for elastic search connector"
+cat << EOF | oc apply -f -
+apiVersion: v1
+kind: Secret
+metadata:
+  namespace: $NAMESPACE
+  name: eei-elastic-credential
+type: Opaque
+stringData:
+  connector.properties: |-
+    dbPassword: $ELASTIC_PASSWORD
+    dbUser: $ELASTIC_USER
```

Comment:
Maybe this is for later (once we get it working), but I think it would make sense to include the hostname here too, which would need to include the ELASTIC_NAMESPACE in it

Reply:
Done